### ISSUE TITLE
RiWordNet does not provide access to sense number

### ISSUE THREAD

**Issue by dhowe** [dhowe/RiTa#2]
@CyrusSUEN Please check if [JWNL](http://jwordnet.sourceforge.net/handbook.html), which RiWordNet uses, can provide this data?

**Comment by CyrusSUEN** [comment]
Do you mean the word sense, one of the meanings of a word?

The closest thing I can find in JWNL API is

public class Word
extends PointerTarget
A Word represents the lexical information related to a specific sense of an IndexWord. * * Word's are linked by Pointers into a network of lexically related words. * getTargets retrieves the targets of these links, and * getPointers retrieves the pointers themselves.

https://web.stanford.edu/class/cs276a/projects/docs/jwnl/javadoc/net/didion/jwnl/data/Word.html

**Comment by dhowe** [comment]
When you get the synsets for a word, each has a sense#. We need to return the synsets in the same order as the sense number. See this example for the word 'dog':

http://wordnetweb.princeton.edu/perl/webwn?c=7&sub=Change&o2=&o0=1&o8=1&o1=1&o7=&o5=&o9=&o6=&o3=&o4=&i=-1&h=00000000&s=dog

There are 7 synsets for ‘dog’, and each has a number (1-7)

**Comment by dhowe** [comment]
First thing to do is verify that we are not already returning them in the correct order...

**Comment by CyrusSUEN** [comment]
Yes, our returning order is getAllSynsets("dog", "n") is different from the result of http://wordnetweb.princeton.edu.

And the order is random when I used the function.
```
1st run: [weenie, domestic dog, pawl, hound, wienerwurst, wiener, hot dog, andiron, frankfurter, hotdog, dog-iron, bounder, click, firedog, blackguard, cad, heel, frank, detent, frump, Canis familiaris]

2nd run: [frump, hot dog, frankfurter, firedog, domestic dog, pawl, wienerwurst, detent, hound, heel, blackguard, andiron, dog-iron, click, Canis familiaris, hotdog, weenie, bounder, cad, wiener, frank]

3rd run: [andiron, Canis familiaris, hound, domestic dog, dog-iron, hot dog, blackguard, detent, frank, heel, hotdog, frankfurter, click, weenie, firedog, bounder, wiener, cad, wienerwurst, pawl, frump]
```

**Comment by dhowe** [comment]
please check it again after setting:

myWordNet.randomizeResults(false);

also, please add randomizeResults to the JSON docs (it is a js-style setter/getter):

myWordNet.randomizeResults(false);  // as setter
boolean val = myWordNet.randomizeResults();  // as getter

**Comment by CyrusSUEN** [comment]
Alright after setting it to false I can the same result everytime I use getAllSynsets() and the result has the same order as the one from http://wordnetweb.princeton.edu/perl/webwn?c=7&sub=Change&o2=&o0=1&o8=1&o1=1&o7=&o5=&o9=&o6=&o3=&o4=&i=-1&h=00000000&s=dog

[domestic dog, Canis familiaris, frump, cad, bounder, blackguard, hound, heel, frank, frankfurter, hotdog, hot dog, wiener, wienerwurst, weenie, pawl, detent, click, andiron, firedog, dog-iron]

**Comment by dhowe** [comment]
Ok, so we have the info. The question is now how to incorporate it into the API. I will take a look...

**Comment by CyrusSUEN** [comment]
> also, please add randomizeResults to the JSON docs (it is a js-style setter/getter):

PR: https://github.com/dhowe/RiTa/pull/135
Verified by generated the reference HTML page, looks good to me.
But perhaps you may want to change the wording a bit.

**Comment by dhowe** [comment]
you didn't update /RiTa/www/reference/RiWordNet.html

**Comment by CyrusSUEN** [comment]
> update /RiTa/www/reference/RiWordNet.html

PR: https://github.com/dhowe/RiTa/pull/136

**Comment by dhowe** [comment]
This is the fix I wanted: 4aaeb1acbd92
